source/components/the-component-life-cycle.md

```diff
@@ -0,0 +1,48 @@
+When a component is invoked, a number of life-cycle hooks are triggered, in sequence, to properly initialize and render the component object and its corresponding DOM element.  The following hooks are a few of the most useful, for current apps.
+
+## init()
```

Comment:
the arrays/objects note is the one I mentioned is useful. as far as the unique ID, I think that is too much implementation detail for the guides